src/java.base/share/classes/jdk/internal/classfile/constantpool/ConstantPool.java

```diff
@@ -25,34 +25,65 @@
 
 package jdk.internal.classfile.constantpool;
 
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 import jdk.internal.classfile.BootstrapMethodEntry;
 import jdk.internal.classfile.ClassReader;
 
 /**
  * Provides read access to the constant pool and bootstrap method table of a
  * classfile.
+ * @jvms 4.4 The Constant Pool
  */
-public sealed interface ConstantPool
+public sealed interface ConstantPool extends Iterable<PoolEntry>
         permits ClassReader, ConstantPoolBuilder {
 
     /**
      * {@return the entry at the specified index}
      *
      * @param index the index within the pool of the desired entry
+     * @throws ConstantPoolException if the index is out of range of the
+     *         constant pool, or is considered unusable
      */
     PoolEntry entryByIndex(int index);
 
     /**
-     * {@return the number of entries in the constant pool}
+     * {@return the size of the constant pool}
```

Comment:
There is still some confusion over the meaning of this method, as "size" (as well as "entry count") could refer to either (a) the number of slots in the constant pool or (b) the number of actual entries in the constant pool, since Constant_{Long,Double} can use two slots.  I agree with the name "size" but we should further clarify that this is the number of slots, but that (a) not all slots necessarily correspond to a valid entry (and therefore entryByIndex may fail) and (b) that iterating the pool may yield fewer entries than the size.

Reply:
I'll mention it in the javadoc